server/public_rpc_server_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 StreamNative, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Additional attributions:
```

Comment:
I found this a bit of an awkward solution to conforming with the go-license check.  This file is about a 50-50 mix of new code and copying from another unit test, so I think this header is accurate but it looks weird.

Dropping the test altogether wouldn't be horrible given it's pretty slow for an unlikely (?) to recur regression.  It took me a bit to track down the issue since I'm not deeply familiar with this code base.  But:

```
10:45:24 > go test -count=1 ./server -run TestWriteClientClose
ok      github.com/streamnative/oxia/server     11.401s
```

Reply:
please run `make license-format`

Reply:
the checks pass.  i'll leave this part alone unless folks want me to drop the test.
```
16:59:41 > make license-format && make license-check && git ls-files -m && echo ok
find . -type f -name '*.go' | grep -v '.pb.go' | xargs go-license --config=.github/license.yml
find . -type f -name '*.go' | grep -v '.pb.go' | xargs go-license --config=.github/license.yml --verify
ok
```